seps/1933-workload-identity-federation.md

```diff
@@ -0,0 +1,228 @@
+# SEP-1933: Workload Identity Federation
+
+- **Status**: Draft
+- **Type**: Standards Track
+- **Created**: 2025-12-05
+- **Author(s)**: Pieter Kasselman
+- **Sponsor**: Darin McAdams
+- **PR**: https://github.com/modelcontextprotocol/modelcontextprotocol/pull/1933
+
+## Abstract
+
+This SEP defines Workload Identity Federation for the Model Context Protocol, enabling autonomous MCP clients to access MCP servers using existing workload identity credentials without requiring separate client registration or long-lived credentials. The proposal leverages the OAuth 2.0 JWT Bearer grant type (RFC 7523) combined with OpenID Connect Discovery to allow MCP clients operating as workloads, such as those deployed in Kubernetes or cloud environments, to present platform-issued JWT credentials directly to authorization servers. This eliminates the need for client secrets, client identifiers, or separate credential management while maintaining security through issuer trust relationships and claim-based authorization.
```

Comment:
I would be really curious to what degree this overlaps with the work captured by [WIMSE](https://datatracker.ietf.org/wg/wimse/documents/). Ideally, we do not invent anything MCP-specific here and instead rely on existing standards, the saem way we did for the core authorization spec.